Issue #867 - UI for including screenshot in bug report

magsout opened the pull request:
Ok, so @miketaylr 

![capture d ecran 2015-12-30 a 21 26 31](https://cloud.githubusercontent.com/assets/1997108/12056877/034fdcda-af3c-11e5-9361-9a89de813ae1.png)

magsout commented on webcompat/templates/form.html:
```diff
@@ -37,19 +37,28 @@
+              <div class="wc-Form-upload-icon" aria-hidden="true">
+                <span class="wc-Icon wc-Icon--cloud-upload"></span>
+              </div>
+              {{ form.image.label(class_='wc-Form-label wc-Form-label--upload') }}
```
If you can remove the attribute `for`  for this `<label>`

miketaylr replied:
I'll see if I can figure that out. ^_^

miketaylr replied:
We might be able to just hand-write the `<label>` and get rid of the python part here (I think). Probably simplest.

miketaylr commented:
Whoa, speedy. :zap:  Will play with this in a bit.

miketaylr commented:
This looks awesome @magsout. I'm gonna add an image preview via `FileReader` (if the browser supports it), otherwise it's hard to know if you've selected something. Once that's in place, I think we can land this and I'll extend it to work with the add-on screenshot stuff (and add drag & drop, etc.)

miketaylr commented:
(that push was just to rebase against master)

miketaylr commented:
![](https://cldup.com/xRwB_5qekV.png)

We can improve the UX in future issues, a couple ideas:

1) ability to clear out file selection, right now you have to just select a new image
2) remove/modify "Add screenshot image" + icon
3) other stuff...

miketaylr commented:
(it can look bad depending on the image, but that's OK I think for now... incremental improvements, etc.):

![](https://cldup.com/HENd9O1S6G.thumb.png)

magsout commented:
@miketaylr we can improve UX before merge if you want. Not a big deal.

miketaylr commented:
@magsout if you have any suggestions, feel free to hack on them. I don't really have any amazing ideas, hence my "let's do it later". :p

When you get a chance, can you review the JS part?

miketaylr commented:
I think I have one idea, let me try to come up with a quick mockup.

miketaylr commented:
Here's one idea, stealing from the original design:

Once we have the upload "read" and the background-image set, we hide the label w/ icon and show a yellow bar on the bottom with a link "Click here to remove image" (I guess it would technically be a button, but I think styling as a link would be good). Once the user clicks that, we remove the bg-image, clear out the file input, and show the original "Upload screenshot image">

We could re-use that same bottom bar for the add-on use case where a user needs to check the checkbox (something like "[ ] Select to include screenshot image")

<img width="671" alt="screen shot 2015-12-31 at 3 33 53 pm" src="https://cloud.githubusercontent.com/assets/67283/12068450/fe7a02d8-afd3-11e5-82d5-445aaad14b07.png">

What do you think?

miketaylr commented:
r? @magsout what do you think?

![](https://cldup.com/qOrYxjD9IG.gif)

magsout commented:
wooot @miketaylr awesome :+1:

magsout commented:
I added a small animation and replaced `<div>` by `<button>`.

I have one issue when I removed image uploaded:

![capture d ecran 2016-01-05 a 12 35 06](https://cloud.githubusercontent.com/assets/1997108/12114264/c659a1a0-b3a8-11e5-8531-79c47d923bb8.png)

miketaylr commented:
Yay, thanks for cleaning up my markup/CSS ^_^. Let me try to reproduce that error, it's kinda wierd.

miketaylr commented:
Ah, the weird view function error is because it's a button -- it's trying to submit or navigate or something. Will sprinkle some JS to fix it, then merge.

miketaylr commented:
> Will sprinkle some JS to fix it

Or just use regular HTML `type=button`.

magsout commented:
Ah yes, I writed `role` instead of type. Good catch @miketaylr
